confmap/expand_test.go

```diff
@@ -317,6 +407,9 @@ func TestResolverExpandStringValues(t *testing.T) {
 
 func newEnvProvider() ProviderFactory {
 	return newFakeProvider("env", func(_ context.Context, uri string, _ WatcherFunc) (*Retrieved, error) {
+		if uri[0:4] != "env:" {
+			uri = "env:" + uri
+		}
```

Comment:
What's the purpose of this? Shouldn't this be handled by the Resolver and not by a provider? It would probably be good to leave a comment to explain it.

Reply:
Comment added